ui/src/components/CheckboxTree/CheckboxTree.tsx

```diff
@@ -0,0 +1,135 @@
+import React, { useEffect, useState, useCallback } from 'react';
+import ColumnLayout from '@splunk/react-ui/ColumnLayout';
+import Button from '@splunk/react-ui/Button';
+import { StyledColumnLayout } from './StyledComponent';
+import {
+    getDefaultValues,
+    getFlattenRowsWithGroups,
+    getNewCheckboxValues,
+    isGroupWithRows,
+} from './CheckboxTree.utils';
+import CheckboxSubTree from './CheckboxSubTree';
+import CheckboxRowWrapper from './CheckboxTreeRowWrapper';
+import { MODE_CREATE } from '../../constants/modes';
+import { CheckboxTreeProps, ValueByField } from './types';
+import { packValue, parseValue } from './utils';
+import { checkValidationForRequired } from './validation';
+
+function CheckboxTree(props: CheckboxTreeProps) {
+    const { field, handleChange, controlOptions, disabled, required } = props;
+    const flattenedRowsWithGroups = getFlattenRowsWithGroups(controlOptions);
+    const shouldUseDefaultValue =
+        props.mode === MODE_CREATE && (props.value === null || props.value === undefined);
+    const initialValues = shouldUseDefaultValue
+        ? getDefaultValues(controlOptions.rows)
+        : parseValue(props.value);
+
+    const [values, setValues] = useState(initialValues);
+
+    useEffect(() => {
+        if (required) {
+            checkValidationForRequired(field, props.label, controlOptions.rows);
+        }
+    }, [required, field, controlOptions.rows, props.label]);
+
+    // Propagate default values on mount if applicable
+    useEffect(() => {
+        if (shouldUseDefaultValue) {
+            handleChange(field, packValue(initialValues), 'CheckboxTree');
+        }
+    }, [field, handleChange, shouldUseDefaultValue, initialValues]);
+
+    const handleRowChange = useCallback(
+        (newValue: { field: string; checkbox: boolean; text?: string }) => {
+            setValues((prevValues: ValueByField) => {
+                const updatedValues = getNewCheckboxValues(prevValues, newValue);
+                handleChange(field, packValue(updatedValues), 'CheckboxTree');
+                return updatedValues;
+            });
+        },
+        [field, handleChange]
+    );
+
+    const handleParentCheckboxTree = useCallback(
```

Comment:
what does `handleParentCheckboxTree` name mean? I am a bit lost of what this method responsible for